src/object_store/src/object/mod.rs

```diff
@@ -318,82 +320,92 @@ impl MonitoredStreamingUploader {
         handle: BoxedStreamingUploader,
         object_store_metrics: Arc<ObjectStoreMetrics>,
         streaming_upload_timeout: Option<Duration>,
+        streaming_upload_finish_timeout: Option<Duration>,
     ) -> Self {
         Self {
             inner: handle,
             object_store_metrics,
             operation_size: 0,
             media_type,
             streaming_upload_timeout,
+            streaming_upload_finish_timeout,
         }
     }
 }
 
 impl MonitoredStreamingUploader {
     pub async fn write_bytes(&mut self, data: Bytes) -> ObjectResult<()> {
-        let operation_type = "streaming_upload_write_bytes";
+        let operation_type = OperationType::StreamingUpload;
+        let operation_type_str = operation_type.as_str();
         let data_len = data.len();
         self.object_store_metrics
             .write_bytes
             .inc_by(data.len() as u64);
         self.object_store_metrics
             .operation_size
-            .with_label_values(&[operation_type])
+            .with_label_values(&[operation_type_str])
             .observe(data_len as f64);
         let _timer = self
             .object_store_metrics
             .operation_latency
-            .with_label_values(&[self.media_type, operation_type])
+            .with_label_values(&[self.media_type, operation_type_str])
             .start_timer();
         self.operation_size += data_len;
 
         let future = async {
             self.inner
                 .write_bytes(data)
-                .verbose_instrument_await("object_store_streaming_upload_write_bytes")
+                .verbose_instrument_await(operation_type_str)
                 .await
         };
         let res = match self.streaming_upload_timeout.as_ref() {
             None => future.await,
             Some(timeout) => tokio::time::timeout(*timeout, future)
                 .await
                 .unwrap_or_else(|_| {
-                    Err(ObjectError::internal(
-                        "streaming_upload write_bytes timeout",
-                    ))
+                    Err(ObjectError::timeout(format!(
+                        "{} timeout",
```

Comment:
Let's include the specific retry and timeout config user can change in the error msg so that user can self-serve when timeout happens.

Reply:
This comment is not resolved.